common/types.go

```diff
@@ -326,25 +326,93 @@ func (b *BidTraceV2WithTimestampJSON) ToCSVRecord() []string {
 	}
 }
 
+type BidTraceV2WithBlobFields struct {
+	builderApiV1.BidTrace
+	BlockNumber   uint64 `db:"block_number"    json:"block_number,string"`
+	NumTx         uint64 `db:"num_tx"          json:"num_tx,string"`
+	NumBlobs      uint64 `db:"num_blobs"       json:"num_blobs,string"`
+	BlobGasUsed   uint64 `db:"blob_gas_used"   json:"blob_gas_used,string"`
+	ExcessBlobGas uint64 `db:"excess_blob_gas" json:"excess_blob_gas,string"`
+}
+
+type BidTraceV2WithBlobFieldsJSON struct {
+	Slot                 uint64 `json:"slot,string"`
+	ParentHash           string `json:"parent_hash"`
+	BlockHash            string `json:"block_hash"`
+	BuilderPubkey        string `json:"builder_pubkey"`
+	ProposerPubkey       string `json:"proposer_pubkey"`
+	ProposerFeeRecipient string `json:"proposer_fee_recipient"`
+	GasLimit             uint64 `json:"gas_limit,string"`
+	GasUsed              uint64 `json:"gas_used,string"`
+	Value                string `json:"value"`
+	NumTx                uint64 `json:"num_tx,string"`
+	BlockNumber          uint64 `json:"block_number,string"`
+	NumBlobs             uint64 `json:"num_blobs,string"`
+	BlobGasUsed          uint64 `json:"blob_gas_used,string"`
+	ExcessBlobGas        uint64 `json:"excess_blob_gas,string"`
+}
+
+func (b BidTraceV2WithBlobFields) MarshalJSON() ([]byte, error) {
```

Comment:
why does it need custom marshal and unmarshal functions?

Reply:
The bid trace is an embedded struct and json.Marshal does not marshal properly for embedded structs.